security-policy.md

```diff
@@ -1,45 +1,55 @@
+> **WARNING:** To ensure end-user security, **please do not use any other medium to report security vulnerabilities**. Also, kindly refrain from disclosing the vulnerability details you come across with others, in any forums, sites, or other groups (public or private) before its mitigation and disclosure processes are completed.
 
-Use the following key to send secure messages to [security@ballerina.io](mailto:security@ballerina.io):
 
-> security@ballerina.io: 0168 DA26 2989 0DB9 4ACD 8367 E683 061E 2F85 C381 [pgp.mit.edu](https://pgp.surfnet.nl/pks/lookup?op=vindex&fingerprint=on&search=0xE683061E2F85C381)
+If you would like, you can encrypt your report using the  following public key to send secure messages to [security@ballerina.io](mailto:security@ballerina.io):
 
-Also, use the following template when reporting vulnerabilities so that it contains all the required information and helps expedite the analysis and mitigation process.
+> security@ballerina.io: AC48 3C56 C0A0 6020 4BBE F3E4 182F 3F21 255F CCE9 
 
-- **Vulnerable Ballerina artifact(s) and version(s):** list of the Ballerina artifacts/versions that are considered as vulnerable
-- **Overview:** high-level overview of the issue and self-assessed severity
+This key can also be found at [keys.openpgp.org](https://keys.openpgp.org/vks/v1/by-fingerprint/AC483C56C0A060204BBEF3E4182F3F21255FCCE9).
+
+Also, use the following template when reporting vulnerabilities so that it contains all the required information that helps expedite the analysis and mitigation process.
+
+- **Vulnerable Ballerina artifact(s) and version(s):** list  the vulnerable Ballerina artifact(s) and version(s) 
+- **Overview:** provide a high-level overview of the issue and self-assessed severity
 - **Description:** include the steps to reproduce
-- **Impact:** self-assessed impact
-- **Solution:** any proposed solution
+- **Impact:** state self-assessed impact
+- **Solution:** propose a  solution if you have one
+
+We will keep you informed of the progress towards a fix and disclosure of the vulnerability if the reported issue is identified as a true positive.
+
+## Handling a Vulnerability
+
+Here is an overview of our approach to handling vulnerabilities:
 
-We will keep you informed of the progress towards a fix and disclosure of the vulnerability if the reported issue is identified as a true positive. 
+1. The vulnerability will be reported privately to our security team at [security@ballerina.io](mailto:security@ballerina.io).
+2. Your email will be acknowledged within 24 hours, and you’ll receive a detailed response to your email indicating the next steps in handling your report. You will be updated on the progress. 
+3. If case of a true-positive finding, the reported vulnerability will be confirmed and fixed by the relevant teams at WSO2.
+4. The fix will be applied to the main branch and a new version of the distribution will be released immediately if required.
```

Comment:
Is it correct to say new version of distribution here? As this is same regarding any component/website etc

Reply:
Yes, it should be changed. How about something like this?
"4. The fix will be applied to the affected components and a new version will be released immediately if required."

Reply:
+1

Reply:
```suggestion
4. The fix will be applied to the affected components and a new version will be released immediately if required.
```